### PR TITLE
Feat!: Expand Functionality of GitHub Action CI/CD Bot

### PR DESCRIPTION
**Pull request by eakmanrq**
Prior to this change the GitHub CI/CD bot was mostly focused on what is considered a "synchronized" code/data deployment pattern. It did have a way to configure it differently but it didn't have first class support. In addition the bot assumed you did not want to auto-categorize changes and backfill models and would error if it encountered a change that hadn't been categorized.

This new version adds support for "desynchronized" deployments and auto-categorization. 

Tests were drastically changed in order to make it easy to check for bot behavior without having to run the bot itself. 

BREAKING CHANGE: The CI/CD bot used to use CLI arguments as a way for configuring behavior. Now this is done entirely from the SQLMesh project config.

### CODE REVIEW

File: sqlmesh/cicd/config.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from pydantic import Field
+
+from sqlmesh.integrations.github.cicd.config import GithubCICDBotConfig
+
+if sys.version_info >= (3, 9):
+    from typing import Annotated
+else:
+    from typing_extensions import Annotated
+
+CICDBotConfig = Annotated[
```
Comment (izeigerman) [inline sqlmesh/cicd/config.py:13]: Can we just make it an alias for now?
```
CICDBotConfig = GithubCICDBotConfig
```

File: sqlmesh/integrations/github/cicd/command.py [hunk 2]
```diff
@@ -166,41 +170,55 @@ def _run_all(
...
+        elif not pr_environment_updated:
+            skip_reason = (
+                "Skipped Deploying to Production because the PR environment was not updated"
+            )
+        elif not prod_plan_generated:
+            skip_reason = (
+                "Skipped Deploying to Production because the production plan could not be generated"
+            )
+        else:
+            skip_reason = "Skipped Deploying to Production for an unknown reason"
```
Comment (izeigerman) [inline sqlmesh/integrations/github/cicd/command.py:212]: Is it even possible to reach this branch with the current implementation?

Comment (eakmanrq) [reply sqlmesh/integrations/github/cicd/command.py:212]: No there is not a known way to reach this in current implementation.

File: docs/integrations/github.md [hunk 3]
```diff
@@ -1,24 +1,17 @@
...
-![Automated Unit Tests with Error Summary](github/github_test_summary.png)
-### Automatically create PR Environments that represent the code changes in the PR
-![Environment Summary](github/github_env_summary.png)
-### Enforce that certain reviewers have approved of the PR before it can be merged
-![Required Approver](github/github_reviewers.png)
-### Preview Prod Plans before Deploying
-![Preview Prod Plans](github/github_prod_plan_preview.png)
-### Automatic deployments to production and merge
-![Deployed Plans](github/github_deployed_plans.png)
+The GitHub Actions CI/CD Bot is enables teams to automate their SQLMesh projects using GitHub Actions. It can be configured to perform the following things:
```
Comment (tobymao) [inline docs/integrations/github.md:3]: Bot enables

**Review by izeigerman — APPROVED** [review]

**Pull request merged**